### ISSUE TITLE
Error while building "Cannot read /build/build"

### ISSUE THREAD

**Issue by wbinek** [asaezper/qt-for-python-rpi3-docker#3]
I have been trying to compile the Pyside2 for Raspberry using your code but every time I try I get the following error:

```
2.1) Configuring...
Creating qmake...
[...]
g++ -o "../bin/qmake" main.o meta.o option.o project.o property.o ioutils.o proitems.o qmakebuiltins.o qmakeevaluator.o qmakeglobals.o qmakeparser.o qmakevfs.o pbuilder_pbx.o makefile.o makefiledeps.o metamakefile.o projectgenerator.o unixmake2.o unixmake.o mingw_make.o msbuild_objectmodel.o msvc_nmake.o msvc_objectmodel.o msvc_vcproj.o msvc_vcxproj.o winmakefile.o xmloutput.o qtextcodec.o qutfcodec.o qendian.o qglobal.o qlogging.o qmalloc.o qnumeric.o qoperatingsystemversion.o qrandom.o qabstractfileengine.o qbuffer.o qdatastream.o qdebug.o qdir.o qdiriterator.o qfile.o qfiledevice.o qfileinfo.o qfilesystemengine.o qfilesystementry.o qfsfileengine.o qfsfileengine_iterator.o qiodevice.o qsettings.o qtemporaryfile.o qtextstream.o qjsonarray.o qjson.o qjsondocument.o qjsonobject.o qjsonparser.o qjsonvalue.o qmetatype.o qsystemerror.o qvariant.o quuid.o qarraydata.o qbitarray.o qbytearray.o qbytearraymatcher.o qcryptographichash.o qdatetime.o qhash.o qlinkedlist.o qlist.o qlocale.o qlocale_tools.o qmap.o qregexp.o qringbuffer.o qstringbuilder.o qstring_compat.o qstring.o qstringlist.o qversionnumber.o qvsnprintf.o qxmlstream.o qxmlutils.o qcore_unix.o qfilesystemengine_unix.o qfilesystemiterator_unix.o qfsfileengine_unix.o qlocale_unix.o  qlibraryinfo.o   -Wl,--gc-sections
Cannot find directory: 
Cannot read /build/build: No such file or directory
Error processing project file: /build
```

This happens when the host is Ubuntu 18.04 in VirtualBox and using modified version of your scripts in windows PowerShell.

Do you know how to solve this issue?

**Comment by asaezper** [comment]
Hi! It works until Qt 5.12.0 (included). From Qt 5.12.1 and above the qmake generation fails. I will investigate the problem.

**Comment by wbinek** [comment]
Hi! Thanks for the answer. 

I have tried with 5.12.0 and it failed for me as well. The error appeared during configuring qt it didn't even reached make step.

Finally I managed to compile pyside2 directly on raspberry pi using this tutorial: https://www.tal.org/tutorials/building-qt-512-raspberry-pi for building Qt and commands from your scripts to build PySide2.

**Comment by asaezper** [comment]
Could you tell me which error do you have with 5.12.0? 
On the other hand, until PySide supports cross-compilation (https://bugreports.qt.io/browse/PYSIDE-802), I guess the best approach is to do a native compilation. Here I only try to emulate it over a Docker to avoid using a real Pi.

**Comment by wbinek** [comment]
I understand. As I said finally I managed to make native compilation although doing it via docker may be easier and faster. There where no compilation errors. At some point during `make configure`
 it just said that it can't find /build/build directory (as mentioned in first message).

At first I thought it may be some problem with path mappings, but I have tried on different systems and the problem persisted.

**Comment by asaezper** [comment]
Which version of Qt do you compile in the Pi? I'm working on a script for compiling Qt and Qt for Python natively in the Pi that also works in the docker, but I'm still getting problems with versions > 5.12.0.

**Comment by wbinek** [comment]
Finally I have used 5.12.3. I followed [www.tal.org/tutorials/building-qt-512-raspberry-pi](https://www.tal.org/tutorials/building-qt-512-raspberry-pi) tutorial to build Qt. To make x11 to work apart from libs from the tutorial I had to install some additional ones mentioned in https://doc.qt.io/qt-5/linux-requirements.html.

My configure command was:
```
PKG_CONFIG_LIBDIR=/usr/lib/arm-linux-gnueabihf/pkgconfig:/usr/share/pkgconfig \
../qt-everywhere-src-5.12.3/configure -platform linux-rpi3-g++ \
-v \
-opengl es2 -eglfs \
-no-gtk \
-opensource -confirm-license -release \
-reduce-exports \
-force-pkg-config \
-nomake examples -no-compile-examples \
-skip qtwayland \
-skip qtwebengine \
-no-feature-geoservices_mapboxgl \
-qt-pcre \
-no-pch \
-ssl \
-evdev \
-system-freetype \
-fontconfig \
-glib \
-prefix /opt/Qt5.12 \
-qpa eglfs \
-skip qtscript
```
I could only get `make` working on one core. Every time I wanted to use more than one core I sooner or later got out of memory error.

Then I used some of the commands from your scripts to build pyside2
```
# Dependencies Qt For Python - Required
sudo apt-get install python3 python3-dev python3-pip python3-venv libxml2 libxml2-dev libxslt1.1 libxslt1-dev cmake
python3 -m pip install setuptools wheel

# On target raspberry install clang6
"deb http://archive.raspbian.org/raspbian buster main contrib non-free rpi firmware" >> /etc/apt/sources.list.d/buster.list
sudo apt update
sudo apt install libc6 libc6-dbg libc6-dev llvm-6.0 libclang-6.0-dev -y
sudo rm /etc/apt/sources.list.d/buster.list
sudo apt update
sudo ln -s /usr/bin/llvm-config-6.0 /usr/bin/llvm-config
LLVM_INSTALL_DIR=/usr/lib/llvm-6.0

# libxslt uses xlocale.h, which is not available if glibc >= 2.26
# https://sourceware.org/glibc/wiki/Release/2.26#Packaging_Changes
sudo sed -i '/#include <xlocale.h>/d' /usr/include/libxslt/xsltlocale.h

# Download pyside2 
QT_VERSION="5.12"
QT_SUBVERSION="5.12.3"
curl -Ls "https://download.qt.io/official_releases/QtForPython/pyside2/PySide2-$QT_SUBVERSION-src/pyside-setup-everywhere-src-$QT_SUBVERSION.tar.xz" | tar -xJf -
cd pyside-setup-everywhere-src-$QT_SUBVERSION


# Workarounds (execute in pyside2 sources directory)
# Linux-ld so in arm
sed -i 's/ld-linux.so.2/ld-linux-armhf.so.3/g' build_scripts/utils.py 
# OpenGLES 2.0 path fixed
sudo sed -i 's/=\/opt\/vc\/include/\/opt\/vc\/include/g' /opt/Qt5.12/lib/cmake/Qt5Gui/Qt5GuiConfigExtras.cmake
sudo ln -s /opt/vc/include/GLES2 /opt/Qt5.12/include/

# Build wheels
python3 setup.py bdist_wheel --qmake=/opt/Qt5.12/bin/qmake --ignore-git --standalone 
```

**Comment by asaezper** [comment]
Hi!

Thanks for your info! After some thinking, I chosed to keep using the "fake cross-compilation" mode (with a patch in the compilation script) instead of using a new native configuration file for Raspberry Pi (the option people from your link used)

I have been working on a new version which is now available. I created:
- A script for the native compilation.
- The docker for executing this script on a regular x96/x64 host

I have tested both, so I hope they work for you if you want to compilate a new version of Qt for Python or whatever.
